### PR TITLE
Minor tidying of API tests files

### PR DESCRIPTION
**Pull request by jhendersonHDF**
Pick off a few of the small suggested changes

### CODE REVIEW

File: test/API/CMakeLists.txt [hunk 1]
```diff
@@ -9,20 +9,6 @@
...
-#------------------------------------------------------------------------------
-set(HDF5_TEST_API_CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMake")
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${HDF5_TEST_API_CMAKE_MODULE_PATH})
-
-# TODO: probably not necessary
-#------------------------------------------------------------------------------
-# Setup CMake Environment
-#------------------------------------------------------------------------------
-if (WIN32)
-  message("The HDF5 API test suite is currently not supported on this platform." FATAL_ERROR)
```
Comment (jhendersonHDF) [inline test/API/CMakeLists.txt]: There shouldn't be any reason that these tests can't be built and run on Windows with the native VOL connector. Support for other VOL connectors on Windows is a different matter.

Comment (byrnHDF) [reply test/API/CMakeLists.txt]: if (HDF5_TEST_PARALLEL)
  set (HDF5_TEST_API_HAVE_PARALLEL 1)
endif ()

I can't find any other us of this HDF5_TEST_API_HAVE_PARALLEL variable.

File: test/API/CMakeLists.txt [hunk 2]
```diff
@@ -9,20 +9,6 @@
 # help@hdfgroup.org.
 #
 
-#------------------------------------------------------------------------------
-# Set module path
```
Comment (byrnHDF) [inline test/API/CMakeLists.txt]: I can take care of these changes in this file

**Review by byrnHDF — APPROVED** [review]

File: test/API/CMakeLists.txt (hunk 1 above)
Comment (jhendersonHDF) [reply test/API/CMakeLists.txt]: Ah yes, that should have been removed when I switched over to checking for H5_HAVE_PARALLEL instead.

**Review by lrknox — APPROVED** [review]

**Pull request merged**